### PR TITLE
Add Pagination

### PR DESCRIPTION
**Pull request by appletan9797**

### CODE REVIEW

**Comment by appletan9797** [comment]
[Apple - Laravel Crawler (add pagination)](https://trello.com/c/nVIES8wb/64-apple-laravel-crawler-add-pagination)

File: app/Http/Controllers/PostController.php [hunk 1]
```diff
@@ -10,8 +10,9 @@ class PostController extends Controller
 {
     public function index()
     {
-        $result = DB::table('posts')->get();
-        return $result->toJson();
+        $result = DB::table('posts')->paginate(5);
+        return response($result,200);
+        //return $result->toJson();
```
Comment (koshuang) [inline app/Http/Controllers/PostController.php:15]: It’s better to remove the comment.

File: resources/js/components/showhistory.js [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
+    const fetchResult = async (pageNumber = 1) => {
         setSpinner(true);
 
-        axios.get('http://localhost/api/results')
+        await axios.get('http://localhost/api/results?page='+pageNumber)
         .then(res => {
             setSpinner(false);
-            setResults(res.data);
+            setPagination(res.data);
+            console.log(res);
```
Comment (koshuang) [inline resources/js/components/showhistory.js:20]: You can put message first to make it more meaningful.

console.log(“api results”, res);

File: resources/js/components/showhistory.js [hunk 3]
```diff
@@ -62,17 +63,31 @@ const ShowHistory = () =>{
...
+            itemsCountPerPage={paginations?.per_page ? paginations?.per_page : 0}
+            totalItemsCount={paginations?.total ? paginations?.total : 0}
+            onChange={(pageNumber) => {fetchResult(pageNumber);}}
+            itemClass="page-item"
+            linkClass="page-link"
+            firstPageText="First Page"
+            lastPageText="Last Page"
+        />
+        }
+    }
```
Comment (koshuang) [inline resources/js/components/showhistory.js:79]: Should have better format:

    const returnPagination = () => {
         if (!spinner) {
             return (
                 <Pagination
                     activePage={paginations?.current_page ? paginations?.current_page : 0}
                     itemsCountPerPage={paginations?.per_page ? paginations?.per_page : 0}
                     totalItemsCount={paginations?.total ? paginations?.total : 0}
                     onChange={(pageNumber) => {fetchResult(pageNumber);}}
                     itemClass="page-item"
                     linkClass="page-link"
                     firstPageText="First Page"
                     lastPageText="Last Page"
                 />
             );
         }
     }

**Review by koshuang — COMMENTED** [review]
For the commit message, you can use what we said for conventional commits which is like:

feat: add pagination

**Pull request merged**